retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/RetrofitMeterIdPrefixFunction.java

```diff
@@ -52,34 +53,58 @@ public static RetrofitMeterIdPrefixFunction of(String name) {
         return builder(name).build();
     }
 
+    /**
+     * Returns a newly created {@link RetrofitClassAwareMeterIdPrefixFunction} with the specified {@code name}
+     * and {@code serviceClass}.
+     */
+    public static RetrofitMeterIdPrefixFunction of(String name, Class<?> serviceClass) {
+        return builder(name, serviceClass).build();
+    }
+
     /**
      * Returns a newly created {@link RetrofitMeterIdPrefixFunctionBuilder} with the specified {@code name}.
      */
     public static RetrofitMeterIdPrefixFunctionBuilder builder(String name) {
         return new RetrofitMeterIdPrefixFunctionBuilder(requireNonNull(name, "name"));
     }
 
+    /**
+     * Returns a newly created {@link RetrofitMeterIdPrefixFunctionBuilder} with the specified {@code name}
+     * and {@code serviceClass}.
+     */
+    public static RetrofitMeterIdPrefixFunctionBuilder builder(String name, Class<?> serviceClass) {
+        return new RetrofitMeterIdPrefixFunctionBuilder(requireNonNull(name, "name"),
+                                                        requireNonNull(serviceClass, "serviceClass"));
+    }
+
     private final String name;
     @Nullable
+    private final String serviceName;
+    @Nullable
     private final String serviceTagName;
     @Nullable
     private final String defaultServiceName;
 
-    RetrofitMeterIdPrefixFunction(String name, @Nullable String serviceTagName,
+    RetrofitMeterIdPrefixFunction(String name,
+                                  @Nullable String serviceTagName,
+                                  @Nullable String serviceName,
                                   @Nullable String defaultServiceName) {
         this.name = name;
-        if (serviceTagName != null && defaultServiceName != null) {
-            this.serviceTagName = serviceTagName;
-            this.defaultServiceName = defaultServiceName;
+        if (serviceName != null || defaultServiceName != null) {
+            this.serviceTagName = firstNonNull(serviceTagName, "service");
+        } else if (serviceTagName != null) {
+            throw new IllegalStateException("If you specify serviceTagName you need " +
+                                            "to specify one of defaultServiceName or serviceName");
         } else {
             this.serviceTagName = null;
-            this.defaultServiceName = null;
         }
+        this.serviceName = serviceName;
+        this.defaultServiceName = defaultServiceName;
     }
 
     @Override
     public MeterIdPrefix activeRequestPrefix(MeterRegistry registry, RequestLog log) {
-        final ImmutableList.Builder<Tag> tagListBuilder = ImmutableList.builderWithExpectedSize(3);
+        final ImmutableList.Builder<Tag> tagListBuilder = ImmutableList.builderWithExpectedSize(2);
```

Comment:
Thanks. 😄

Reply:
It is trivial, however, if we want to optimize this:

```java
int expectedSize = serviceTagName != null ? 3 : 2;
ImmutableList.Builder<Tag> tagListBuilder = ImmutableList.builderWithExpectedSize(expectedSize);
```

Reply:
It's only two tags in this case, so I think it's `2`. 😄

Reply:
I misread the code.